File: 02_rookie/week11/121_best-time-to-buy-and-sell-stock_watermelon.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max_profit = 0 # 최대 이익
        min_price = prices[0] # 최소 가격

        for price in prices:
            # 최소 가격 갱신
            if price < min_price:
                min_price = price
                continue
            
            # 최대 이익 갱신
            if max_profit < price - min_price:
                max_profit = price - min_price
        
        return max_profit

File: 02_rookie/week11/121_best-time-to-buy-and-sell-stock_watermelon.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let minPrice = prices[0];
    let result = 0;

    prices.forEach(price => {
        if (price < minPrice) {
            minPrice = price;
            return;
        }
        if (result < price - minPrice) {
            result = price - minPrice;
        }
    })

    return result;
};
